delete mega-move, make non-interactive, enhance data cleaning

Fligel-Kuzminskikh opened the pull request.

Fligel-Kuzminskikh committed:
delete mega-move, make non-interactive, enhance data cleaning

maslinych commented on scripts/illustrators_network.R:
```diff
@@ -0,0 +1,310 @@
+      data$X2 = str_replace_all(data$X2, " � ", "�")
+      if(authors){
+      data$X2 = str_replace_all(data$X2, "�-", "�")
+      pattern4 = or(START %R% "���� ", START %R% "����. ", START %R% "���� � �����. ", START %R% "� ��������� ")
```
кодировка в исходном файле cp1251: Ñîñò Ñîñò è îáðàá. Â îáðàáîòêå  etc. В таком виде смержить не могу

Fligel-Kuzminskikh replied:
каждая cyrillic строка теперь аргумент функции enc2ut8() 
```suggestion
      pattern4 = or(START %R% enc2utf8("���� "), START %R% enc2utf8("����. "), START %R% enc2utf8("���� � �����. "), START %R% enc2utf8("� ��������� "))
```

Fligel-Kuzminskikh replied:
вышеупомянутая функция просто возвращает строку в кодировке UTF-8

maslinych replied:
> каждая cyrillic строка теперь аргумент функции enc2ut8()

нет, Рома, такой подход не годится. Нужно вместо этого просто перекодировать исходный код (файл) в кодировку UTF-8 и следить за тем, чтобы входные данные тоже были в этой кодировке.

maslinych commented on scripts/illustrators_network.R:
```diff
@@ -0,0 +1,310 @@
+library(rebus)
+library(igraph)
+library(tidytext)
+library(optparse)
```
stringr не загружен, а используется в коде

Fligel-Kuzminskikh replied:
дело в том, что пакет stringr включен в библиотеку tidyverse

Fligel-Kuzminskikh replied:
так что отдельной загрузки не требуется

maslinych commented on scripts/illustrators_network.R:
```diff
@@ -0,0 +1,310 @@
+library(optparse)
+
+tidy = function(data, authors=F){
+      data$X2 = str_remove_all(data$X2, "[,]|[.]")
```
Почему Вы избегаете использования %>% pipe? 
data %>% 
  mutate(str_remove_all, X2, "[,.]") %>% 
  mutate(....)

maslinych replied:
X1, X2 в качестве имен колонок это какая-то недотыкомка, нужно было дать нормальные имена колонкам при загрузке данных read_tsv(col_names = c("author", ...)...)

maslinych commented on scripts/illustrators_network.R:
```diff
@@ -0,0 +1,310 @@
+      data$X2 = str_replace_all(data$X2, " � ", "�")
+      if(authors){
+      data$X2 = str_replace_all(data$X2, "�-", "�")
+      pattern4 = or(START %R% "���� ", START %R% "����. ", START %R% "���� � �����. ", START %R% "� ��������� ")
```
pattern4 — плохое (неинформативное) название переменной

maslinych commented on scripts/illustrators_network.R:
```diff
@@ -0,0 +1,310 @@
+      bugs = str_extract_all(data$X1, pattern = "[[:digit:]]", simplify = T)
+      bugs = as.data.frame(bugs)
+      bugs = unite(bugs, col = "Work", sep = "", remove = T)
+      for(i in 1:nrow(data)){
```
тут надо избавиться от for, но я не очень понимаю, что Вы хотели здесь сделать

Fligel-Kuzminskikh replied:
здесь я переношу баги(индексы изданий в колонке с авторами) непосредственно в колонку с индексами

maslinych commented on scripts/illustrators_network.R:
```diff
@@ -0,0 +1,310 @@
+}
+
+int_extract = function(cleaned.data){
+pattern = one_or_more('[[:digit:]]') %R%
```
весь код внутри функций должен быть оформлен с indent'ом

maslinych commented on scripts/illustrators_network.R:
```diff
@@ -0,0 +1,310 @@
+library(tidytext)
+library(optparse)
+
+tidy = function(data, authors=F){
```
codestyle, которому я следую, требует использовать оператор <- для присвоения значений переменных, а не =
Перед фигурной скобкой с телом функции нужно ставить пробел

maslinych replied:
вместо T и F нужно использовать полные TRUE и FALSE

maslinych commented on scripts/illustrators_network.R:
```diff
@@ -0,0 +1,310 @@
+
+main = function(args){
+  setwd(args$outdir)
+  illus = read_tsv(args$inillus, col_names = F)
```
содержание main() поддается дальнейшему абстрагированию, потому что тут в теле повторяются серии однотипных последовательностей действий для нескольких видов сетей, каждую такую серию можно описать как функцию, а тип сети передавать как параметр

maslinych reviewed (commented):
Код стал значительно лучше и читабельнее, но все равно есть еще большое пространство для улучшения (но это как всегда). Главный блокер, который не дает пока возможности мержить — это кодировка файла. Плюс я написал еще несколько комментариев по оформлению и структуре кода — это не все, что нужно исправить, но обозначение направления, куда нужно двигаться.

Fligel-Kuzminskikh committed:
consider all the comments, but the last one 

include pipes, rename variables, use TRUE and FALSE, adapt to the coding style, use indents
